io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -594,6 +605,11 @@ pub trait LvsLvol: LogicalVolume + Share {
     /// Wrapper function to destroy replica and its associated snapshot if
     /// replica is identified as last clone.
     async fn destroy_replica(mut self) -> Result<String, Error>;
+
+    /// Resize a replica. The resize can be expand or shrink, depending
+    /// upon if required size is more or less than current size of
+    /// the replpica.
+    async fn resize_replica(mut self, resize_to: u64) -> Result<Lvol, Error>;
```

Comment:
Why move and return Lvol?

Reply:
What would be more idiomatic way? Perhaps in this function we can take a `&self` and return only `Result<(), Error)` ?